core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -627,6 +641,29 @@ private[spark] class ExecutorAllocationManager(
     def isExecutorIdle(executorId: String): Boolean = {
       !executorIdToTaskIds.contains(executorId)
     }
+
+    /**
+     * Get the number of locality aware pending tasks and related locality preferences as the
```

Comment:
Change this to:
Get a tuple of (the total number of pending tasks with locality preferences, a map where each pair is a node and the number of pending tasks that would like to be scheduled on that node).